### PR TITLE
Refactoring authenticate middleware and userManagement util 

### PR DESCRIPTION
**Pull request by virtser**
connects to https://github.com/dorbel-tech/dorbel-app/issues/949

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "release": "scripts/release.sh",
     "test": "yarn run test:unit",
-    "test:unit": "nyc --all mocha --timeout 10000 test/before-tests.js test/**/*.spec.js",
+    "test:unit": "nyc --all mocha --timeout 10000 test/before-tests.js test/**/*.spec.js ",
```
Comment (amoslande) [inline package.json]: ?

File: src/utils/user/permissions.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+function isUserAdmin(user) {
+  return user && user.role === 'admin';
+}
+
+function isResourceOwner(user, resourceOwnerId) {
+  return user && user.id === resourceOwnerId;
+}
+
+function isResourceOwnerOrAdmin(user, resourceOwnerId) {
```
Comment (lxngxr) [inline src/utils/user/permissions.js]: WDYT about having this as the only `isResourceOwner` method?

Comment (virtser) [inline src/utils/user/permissions.js]: What do you mean?

File: package.json (hunk 1 above)
Comment (virtser) [inline package.json]: fixed

**Pull request merged**